keyboards/matrix/m12og/m12og.c

```diff
@@ -0,0 +1,32 @@
+/**
+ * m12og.c
+ */
+
+#include "m12og.h"
+
+void matrix_init_user(void) {
+	setPinOutput(C6);
+	setPinOutput(B2);
+	setPinOutput(B1);
+}
+
+void led_set_user(uint8_t usb_led) {
+
+	if (usb_led & (1 << USB_LED_NUM_LOCK)) {
+        writePinLow(B1);
+	} else {
+        writePinHigh(B1);
+	}
+
+	if (usb_led & (1 << USB_LED_CAPS_LOCK)) {
+        writePinLow(C6);
+	} else {
+        writePinHigh(C6);
+	}
+
+	if (usb_led & (1 << USB_LED_SCROLL_LOCK)) {
+        writePinLow(B2);
+	} else {
+        writePinHigh(B2);
+	}
+}
```

Comment:
Could you update this to use the newer functionality?  

```c
bool led_update_kb(led_t led_state) {
    bool res = led_update_user(led_state);
    if(res) {
        // writePin sets the pin high for 1 and low for 0.
        // In this example the pins are inverted, setting
        // it low/0 turns it on, and high/1 turns the LED off.
        // This behavior depends on whether the LED is between the pin
        // and VCC or the pin and GND.
        writePin(B1, !led_state.num_lock);
        writePin(C6, !led_state.caps_lock);
        writePin(B2, !led_state.scroll_lock);
    }
    return res;
}
```

This allows users to override the functionality in a way that prevents flicker or weird behavior.

Reply:
updated, seems I'm a bit out of date about the qmk codes :)

Reply:
No worries. This is actually very new (within the last month or two), so it's understandable.

However it's a very good change, and we want to help with it's adoption :)